build failure on intelAMT provider

Build fails with the Intel AMT provider, the cause seems to be the import overrides https://github.com/bmc-toolbox/bmclib/blob/main/go.mod#L38 

To reproduce this issue,
 1. Create a new go project directory `mkdir foo && cd foo && go mod init foo`
 2. Copy over a bmclib example - https://github.com/bmc-toolbox/bmclib/blob/main/examples/inventory/main.go
 3. Run `go mod tidy`
 4. Get the latest bmclib/v2 `go get github.com/bmc-toolbox/bmclib/v2@4400298`
 5. Run go build `go build .`

The below error will be returned.
```
# github.com/bmc-toolbox/bmclib/v2/providers/intelamt
/Users/jrebello/go/pkg/mod/github.com/bmc-toolbox/bmclib/v2@v2.0.1-0.20221122065736-4400298194c3/providers/intelamt/intelamt.go:61:3: unknown field 'Logger' in struct literal of type amt.Connection
/Users/jrebello/go/pkg/mod/github.com/bmc-toolbox/bmclib/v2@v2.0.1-0.20221122065736-4400298194c3/providers/intelamt/intelamt.go:70:11: cannot use client (variable of type *amt.Client) as type amtProvider in struct literal:
        *amt.Client does not implement amtProvider (wrong type for IsPoweredOn method)
                have IsPoweredOn() (bool, error)
                want IsPoweredOn(context.Context) (bool, error)
/Users/jrebello/go/pkg/mod/github.com/bmc-toolbox/bmclib/v2@v2.0.1-0.20221122065736-4400298194c3/providers/intelamt/intelamt.go:91:4: unknown field 'Logger' in struct literal of type amt.Connection
/Users/jrebello/go/pkg/mod/github.com/bmc-toolbox/bmclib/v2@v2.0.1-0.20221122065736-4400298194c3/providers/intelamt/intelamt.go:97:14: cannot use client (variable of type *amt.Client) as type amtProvider in assignment:
        *amt.Client does not implement amtProvider (wrong type for IsPoweredOn method)
                have IsPoweredOn() (bool, error)
                want IsPoweredOn(context.Context) (bool, error)
```

The error goes away if the project importing bmclib/v2 includes these lines in go.mod
```
replace github.com/ammmze/go-amt => github.com/jacobweinstock/go-amt v0.0.0-20221115190754-36c39f21b864
replace github.com/ammmze/wsman => github.com/jacobweinstock/wsman v0.0.0-20221115191137-e06ce6a5341d
```

@jacobweinstock

--- a/.github/mergify.yml
+++ b/.github/mergify.yml
@@ -8,7 +8,7 @@ queue_rule:
 pull_request_rules:
   - name: Automatic merge on approval
     conditions:
-      - base=v2
+      - base=main
       - "#approved-reviews-by>=1"
       - "#changes-requested-reviews-by=0"
       - check-success='lint'

--- a/go.mod
+++ b/go.mod
@@ -3,12 +3,12 @@ module github.com/bmc-toolbox/bmclib/v2
 go 1.18
 
 require (
-	github.com/ammmze/go-amt v0.0.4
 	github.com/bmc-toolbox/common v0.0.0-20221115135648-0b584f504396
 	github.com/bombsimon/logrusr/v2 v2.0.1
 	github.com/go-logr/logr v1.2.3
 	github.com/google/go-cmp v0.5.8
 	github.com/hashicorp/go-multierror v1.1.1
+	github.com/jacobweinstock/go-amt v0.0.0-20221125040441-53475f4ae023
 	github.com/jacobweinstock/registrar v0.4.6
 	github.com/pkg/errors v0.9.1
 	github.com/sirupsen/logrus v1.8.1
@@ -22,9 +22,9 @@ require (
 require (
 	github.com/VictorLowther/simplexml v0.0.0-20180716164440-0bff93621230 // indirect
 	github.com/VictorLowther/soap v0.0.0-20150314151524-8e36fca84b22 // indirect
-	github.com/ammmze/wsman v0.0.2 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/hashicorp/errwrap v1.1.0 // indirect
+	github.com/jacobweinstock/wsman v0.0.0-20221125035617-2eae65734c77 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/satori/go.uuid v1.2.0 // indirect
 	golang.org/x/lint v0.0.0-20210508222113-6edffad5e616 // indirect
@@ -34,7 +34,3 @@ require (
 	golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
-
-replace github.com/ammmze/go-amt => github.com/jacobweinstock/go-amt v0.0.0-20221115190754-36c39f21b864
-
-replace github.com/ammmze/wsman => github.com/jacobweinstock/wsman v0.0.0-20221115191137-e06ce6a5341d

--- a/providers/intelamt/intelamt.go
+++ b/providers/intelamt/intelamt.go
@@ -6,9 +6,9 @@ import (
 	"strconv"
 	"strings"
 
-	"github.com/ammmze/go-amt"
 	"github.com/bmc-toolbox/bmclib/v2/providers"
 	"github.com/go-logr/logr"
+	"github.com/jacobweinstock/go-amt"
 	"github.com/jacobweinstock/registrar"
 )
 

--- a/providers/intelamt/intelamt_test.go
+++ b/providers/intelamt/intelamt_test.go
@@ -5,10 +5,10 @@ import (
 	"errors"
 	"testing"
 
-	"github.com/ammmze/go-amt"
 	"github.com/go-logr/logr"
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/jacobweinstock/go-amt"
 )
 
 type mock struct {

